src/pip/_internal/cli/req_command.py

```diff
@@ -438,6 +438,9 @@ def get_requirements(
                     isolated=options.isolated_mode,
                     use_pep517=options.use_pep517,
                     user_supplied=True,
+                    config_settings=parsed_req.options.get("config_settings")
+                    if parsed_req.options
+                    else None,
```

Comment:
Is the `if` part to guard against None? (because the `get` part already handles the empty dict part)

Reply:
but parsed_req.options could be None
```python
class ParsedRequirement:
    def __init__(
        self,
        requirement: str,
        is_editable: bool,
        comes_from: str,
        constraint: bool,
        options: Optional[Dict[str, Any]] = None,
        line_source: Optional[str] = None,
        ....
```